torch/_dynamo/variables/dicts.py

```diff
@@ -221,18 +221,23 @@ def unpack_var_sequence(self, tx):
         return result
 
     @classmethod
-    def get_key(cls, arg: VariableTracker):
+    def get_key(cls, tx, arg: VariableTracker):
         if isinstance(arg, TensorVariable) and arg.specialized_value is not None:
             return arg.specialized_value
+        elif isinstance(arg, variables.NNModuleVariable):
+            return tx.output.nn_modules[arg.module_key]
         else:
             return arg.as_python_constant()
 
     @classmethod
     def is_valid_key(cls, key):
         return (
             key.is_python_constant()
-            or (isinstance(key, TensorVariable) and key.specialized_value is not None)
-            or (isinstance(key, ConstantVariable) and key.python_type() is torch.dtype)
+            or isinstance(key, TensorVariable)
```

Comment:
You probably want to add parens around the ands here, the order of operation is a bit counter-intuitive otherwise (and goes before or in Python)